### PR TITLE
fix: check cycle detection with exclusion

### PR DESCRIPTION
**Pull request by miparnisari**
## Description
Fixing a panic that occurs on some cases where the model has cycles and exclusion is involved.

## References

The panic was introduced in https://github.com/openfga/openfga/pull/1372

### CODE REVIEW

File: internal/graph/cycle_check_resolver_test.go [hunk 1]
```diff
@@ -60,3 +66,56 @@ func TestCycleDetectionCheckResolver(t *testing.T) {
...
+
+	typesys, err := typesystem.NewAndValidate(
+		context.Background(),
+		model,
+	)
+	require.NoError(t, err)
+
+	ctx := storage.ContextWithRelationshipTupleReader(context.Background(), ds)
+	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+	_, err = cycleDetectionCheckResolver.ResolveCheck(ctx, &ResolveCheckRequest{
```
Comment (jon-whit) [inline internal/graph/cycle_check_resolver_test.go:114]:
```suggestion
	resp, err = cycleDetectionCheckResolver.ResolveCheck(ctx, &ResolveCheckRequest{
```

File: internal/graph/cycle_check_resolver_test.go [hunk 2]
```diff
@@ -60,3 +66,56 @@ func TestCycleDetectionCheckResolver(t *testing.T) {
...
+		StoreID:              storeID,
+		AuthorizationModelID: model.GetId(),
+		TupleKey:             tuple.NewTupleKey("group:1", "blocked", "user:jon"),
+		RequestMetadata:      NewCheckRequestMetadata(25),
+	})
+	require.NoError(t, err)
```
Comment (jon-whit) [inline internal/graph/cycle_check_resolver_test.go:120]:
```suggestion
	require.NoError(t, err)
	require.False(t, resp.GetAllowed())
```

Comment (miparnisari) [reply internal/graph/cycle_check_resolver_test.go:121]: I get `allowed=true` 😬

Comment (jon-whit) [reply internal/graph/cycle_check_resolver_test.go:121]: Wait a second, this should be an error `ErrCycleDetected`. Sorry, I stand corrected.

If you start at `group#blocked` --> `group#member` and `group#member` depends on `group#blocked`, then that's a cycle and you cannot resolve. 

So this should be an `ErrCycleDetected`.

Comment (miparnisari) [reply internal/graph/cycle_check_resolver_test.go:121]: should we tackle this in a different PR?

**Review by jon-whit — APPROVED** [review]

**Pull request merged**